gremlin-dotnet/src/Gremlin.Net/Process/Traversal/GraphTraversal.cs

```diff
@@ -141,73 +159,73 @@ private GraphTraversal(ICollection<ITraversalStrategy> traversalStrategies, Byte
         /// <summary>
         ///     Adds the aggregate step to this <see cref="GraphTraversal{SType, EType}" />.
         /// </summary>
-        public GraphTraversal<S, E> Aggregate (Scope scope, string sideEffectKey)
+        public GraphTraversal<S, E_> Aggregate (Scope scope, string sideEffectKey)
         {
             Bytecode.AddStep("aggregate", scope, sideEffectKey);
-            return Wrap<S, E>(this);
+            return Wrap<S, E_>(this);
```

Comment:
ugh - that's too bad....

Reply:
Class was defined as `public class GraphTraversal<S, E> : DefaultTraversal<S, E>`
[C# don't allow to have method with same name as type](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0102)

Possible solutions:
1. rename `E` in class definition (internal effect only, I used it)
2. rename `E` function

Reply:
yeah, i was just making an observation. not much we can do there. personally, i don't mind the underscore but maybe @FlorianHockmann or others could offer a better suggestion?

Reply:
The C# [naming convention suggests](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/generic-type-parameters#type-parameter-naming-guidelines) that names of generic type parameters like these should start with a _T_. We didn't follow that and instead just reused the names from Java _S_ and _E_, but I'm wondering if we should simply change these into _TS_ and _TE_ or even _TStart_ and _TEnd_ to be more descriptive and comply with the naming convention that most C# devs are used to.
This would then also solve this problem here.

If you don't have any objections, then I'll create a separate issue in Jira to change these names in a different PR. But this shouldn't block this PR here. So, I suggest that we keep it for now as you've did in this PR with `E_` and then we can later change that to `TE` / `TEnd` or whatever.

That's definitely better than changing the name of the step `E()` itself as that would lead to a lot more confusion in my opinion.

Reply:
@vkagamlyk would you mind taking the extra step to fix it on this PR? or an immediate follow-up PR? i'd hate for us to forget this change when we have a chance to instituted it on 3.7.x and let this release with the underscore.

Reply:
@FlorianHockmann is it better to rename TStart/TEnd in all affected classes or only fix `GraphTraversal<TStart, TEnd >` in this PR?

Reply:
I would do it in all affected classes to stay consistent.

Reply:
renamed in https://github.com/apache/tinkerpop/pull/1808/commits/fc3480cc5a217a215e48ae13396082effd59085a

Reply:
Should I propagate these changes to `3.5-dev` and `3.6-dev`?